typescript/lib/parsers/openai.ts

```diff
@@ -625,11 +633,10 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
     }
 
     if (output.output_type === "execute_result") {
-      const message = output.data as Chat.ChatCompletionMessageParam;
-      if (message.content != null) {
-        return message.content;
-      } else if (message.function_call) {
-        return JSON.stringify(message.function_call);
+      if (typeof output.data === "string") {
+        return output.data;
+      } else if (output.metadata?.function_call) {
+        return JSON.stringify(output.metadata?.function_call);
```

Comment:
This logic update will break existing aiconfigs. Since this library has already been released, we need to be explicit about breaking changes before making them. As such, it's best to keep the existing logic in addition to your update.

i.e.

```
if (typeof output.data === "string") {
  return output.data;
} else if (output.metadata?.function_call) {
  return JSON.stringify(output.metadata?.function_call);
} else {
  const message = output.data as Chat.ChatCompletionMessageParam;
  if (message?.content != null) {
    return message.content;
  } else if (message.function_call) {
    return JSON.stringify(message.function_call);
  } else {
    return ""
  }
```

Reply:
@saqadri wouldn't that be addressed by a major version bump?

Reply:
Until we do a major version bump, we have to support both. And even when we do a major version bump, ideally we enable a "conversion" in this code path. Can discuss separately, but right now it seems like we introduced a breaking change implicitly